addons/point_of_sale/static/src/js/screens.js

```diff
@@ -1920,8 +1931,16 @@ var PaymentScreenWidget = ScreenWidget.extend({
     click_delete_paymentline: function(cid){
         var lines = this.pos.get_order().get_paymentlines();
         for ( var i = 0; i < lines.length; i++ ) {
-            if (lines[i].cid === cid) {
-                this.pos.get_order().remove_paymentline(lines[i]);
+            var line = lines[i];
+            if (line.cid === cid) {
+                // If an paymentline with a payment terminal linked to
+                // it is removed, the terminal should get a cancel
+                // request.
+                if (line.get_payment_status() && ['waitingCard', 'timeout'].includes(lines[i].get_payment_status())) {
```

Comment:
I think we should also cancel the payment if the status is `waiting`

Reply:
Yeah, I suppose. Normally Adyen will set a different state after the initial request but if the response gets lost I guess we should still do a cancel. Better safe than sorry anyway, cancelling a non-existent transaction doesn't cause any harm.